### PR TITLE
adding loading component

### PR DESCRIPTION
**Pull request by JorgeGarciaIrazabal**
adding loading component
adding chat page (structure)
created develop user to be used when testing

### CODE REVIEW

File: backend/scripts/add_data_base_basic_data.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import json
```
Comment (JorgeGarciaIrazabal) [inline backend/scripts/add_data_base_basic_data.py:1]: just a simple script to include basic data in the database (now only adding the develop user)

File: src/components/Loading.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import React, {Component, PropTypes} from 'react';
```
Comment (JorgeGarciaIrazabal) [inline src/components/Loading.js:1]: loading component (need to be added to every page)

image

the screenShot doesn't move but it is a gif :wink:

File: src/models/model.js [hunk 3]
```diff
@@ -10,7 +10,7 @@ export default class Model {
   @observable updatedAt;
 
   constructor() {
-    this.createdAt = moment().utc().date();
+    this.createdAt = moment().utc().toDate();
```
Comment (JorgeGarciaIrazabal) [inline src/models/model.js:13]: date was getting the day and not the `Date` object

File: src/pages/chat.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import React, {Component, PropTypes} from 'react';
```
Comment (JorgeGarciaIrazabal) [inline src/pages/chat.js:1]: Structure of the `chat` page (nothing in yet)

File: src/pages/login.js [hunk 5]
```diff
@@ -5,33 +5,65 @@ import FontAwesomeIcon from 'react-native-vector-icons/FontAwesome';
...
     try {
+      await this.setSelf(user);
+      this.props.navigator.replace(routes.chat)
+    }
+    catch (e) {
+      console.error(e);
+    }
+  }
+
+  async redirectIfLoggedIn() {
```
Comment (JorgeGarciaIrazabal) [inline src/pages/login.js:41]: redirecting if logged (by default user will be logged in if we are on `__DEV__`)

File: src/routes.js [hunk 6]
```diff
@@ -1,8 +1,13 @@
 import Login from './pages/login';
```
Comment (JorgeGarciaIrazabal) [inline src/routes.js:1]: This is how you add a new page

File: src/pages/login.js [hunk 7]
```diff
@@ -5,33 +5,65 @@ import FontAwesomeIcon from 'react-native-vector-icons/FontAwesome';
...
+    if (__DEV__) {
+      let user = new User();
+      user.email = 'develop+develop@gmail.com';
+      user.password = '__develop';
+      await this.setSelf(user);
+      this.props.navigator.replace(routes.chat)
```
Comment (JorgeGarciaIrazabal) [inline src/pages/login.js:47]: to change the page, just `replace` or `push` from navigator

File: src/routes.js (hunk 6 above)
Comment (Fdom92) [reply src/routes.js:1]: i did this with contact page, now ill get conflicts :D

File: src/pages/login.js (hunk 5 above)
Comment (Fdom92) [reply src/pages/login.js:41]: i was redirecting too once the user login so we need to set up the navigation flow maybe :D

**Review by Fdom92 — APPROVED** [review]

File: src/pages/login.js (hunk 5 above)
Comment (JorgeGarciaIrazabal) [reply src/pages/login.js:41]: the way is to redirect to contacts so you can change it to use yours, no problem.

File: src/routes.js (hunk 6 above)
Comment (JorgeGarciaIrazabal) [reply src/routes.js:1]: come on!!! just a simple rebase, not a big deal :stuck_out_tongue_closed_eyes:

**Pull request merged**